### PR TITLE
Use "command" not "tool" or "utility"

### PR DESCRIPTION
**Pull request by richsalz**
I think this might be doing too much.

I would be happy to remove the find-doc-nits change, and roll back any other editing that wasn't strictly necessary to say "they are called commands"

### CODE REVIEW

**Comment by levitte** [comment]
Er... What's wrong with "utility" and "tool"?

**Comment by richsalz** [comment]
er, what's wrong with consistency?  Why is "openssl asn1" a utility and "openssl rsa" a tool?

**Comment by levitte** [comment]
That didn't answer my question. You could have changed "tool" to "utility" or vice versa, where applicable...

**Comment by richsalz** [comment]
>That didn't answer my question.

Oh, sorry.  Here's the sub-thread in another PR to provide context.  My mistake for not posting this. https://github.com/openssl/openssl/pull/11113#discussion_r380411134

**Comment by levitte** [comment]
For some reason, I find "command" ugly and non-descript.
I noticed the statistics, but I think that might be biased by changes you have made. What happens if you run the same stats script in 1.1.1?

**Comment by richsalz** [comment]
The results are biased because, over a series of a dozen-plus PR's, I worked on the manpages. :) As for why command, I'll defer to this: https://www.kernel.org/doc/man-pages/ and point to one of your favorite sites, http://man7.org/linux/man-pages/man7/man-pages.7.html  (Note that "programs" is not appropriate because we only have one executable.)

As for ugly and nondescript, there are far more uglier things in OpenSSL :)

As for 1.1.1:
```
; cd doc/man1
; g grep command | wc
    234    2286   16992
; g grep utility | wc
     52     468    3460
; g grep tool | wc
     19     182    1306
; 
```

**Comment by mspncp** [comment]
FWIW: If it was my choice, I‘d pick „command“.

**Comment by mspncp** [comment]
> For some reason, I find "command" ugly and non-descript.

For what reason? For me the term fits nicely to other terms like "command line interface (CLI)" or "shell command".  Maybe the comprise would be to call `openssl` a *tool* and `openssl pkey`, `openssl x509`, ... its *commands*.

**Comment by levitte** [comment]
> For what reason?

It's entirely subjective, so my reason is my taste.  Did you expect a logical explanation?

**Comment by levitte** [comment]
BTW,  @richsalz, doc-nits complains about openssl-x509.pod

**Comment by richsalz** [comment]
Yes, the complaint about x509 is deliberate to force a conversation :)
```
doc/man1/openssl-x509.pod:1: found 'utility' should use 'command'
```

I edited many man1 pod pages to get rid of all reference to utility or tool, and made find-doc-nits complain.  Did that go too far?  Comments?

**Comment by levitte** [comment]
I have nothing against a uniform noun for the sub-commands. Whether find-doc-nits should be strict about it, I mostly shrug...

**Comment by levitte** [comment]
Regarding choice of words, I changed my mind, "command" is fine.

It was when I eyed `pkeyutl` and thought back about this PR that I realized that "the pkeyutl utility" is a tautology, and gets a bit silly that way (like when some people talk about "a CD disk").

**Comment by richsalz** [comment]
travis failure is a time-out.  this could be reviewed.

**Comment by richsalz** [comment]
This now fixes #11233 so that POD syntax errors/warnings will be caught in our doc-nits filter.

File: doc/man1/openssl-asn1parse.pod [hunk 1]
```diff
@@ -26,7 +26,7 @@ B<openssl> B<asn1parse>
 
 =head1 DESCRIPTION
 
-This command is a diagnostic utility that can parse ASN.1 structures.
+This command is a diagnostic aide that can parse ASN.1 structures.
```
Comment (InfoHunter) [inline doc/man1/openssl-asn1parse.pod]: As a non-native English speaker, I find the word 'aide' not that frequently used, is it better to replace it with 'helper'?

File: doc/man1/openssl-s_client.pod.in [hunk 2]
```diff
@@ -139,7 +139,7 @@ B<openssl> B<s_client>
 
 This command implements a generic SSL/TLS client which
 connects to a remote host using SSL/TLS. It is a I<very> useful diagnostic
-tool for SSL servers.
+aide for SSL servers.
```
Comment (InfoHunter) [inline doc/man1/openssl-s_client.pod.in]: 'helper'?

File: doc/man1/openssl-asn1parse.pod (hunk 1 above)
Comment (mattcaswell) [reply doc/man1/openssl-asn1parse.pod]: Drive by comment: I think "aide" here is not quite correct anyway. An aide is usually a person. I think this should be a "diagnostic aid"....or just use a different word as @InfoHunter suggests :-)

Comment (InfoHunter) [reply doc/man1/openssl-asn1parse.pod]: Haha, this my first time to encounter the word 'aide', very impressive anyway.

Comment (richsalz) [reply doc/man1/openssl-asn1parse.pod]: Looking at https://www.thesaurus.com/browse/tool and other references, I couldn't find anything better.  And Matt is of course right, "aide" should be "aid"  Argh.  Fixup commit pushed.  Helper doesn't quite convey the right meaning to me.  If folks want it changed -- or they want me to remove the find-doc-nits check (see ```man1words``` in that script), let me know.

Comment (t8m) [reply doc/man1/openssl-asn1parse.pod]: What's wrong with using tool at this place? The word aid looks awkward here to me (although of course I am not native English speaker).
I would remove the check for the tool word.

File: doc/man1/openssl-s_client.pod.in [hunk 3]
```diff
@@ -139,7 +139,7 @@ B<openssl> B<s_client>
 
 This command implements a generic SSL/TLS client which
 connects to a remote host using SSL/TLS. It is a I<very> useful diagnostic
-tool for SSL servers.
+aid for SSL servers.
```
Comment (t8m) [inline doc/man1/openssl-s_client.pod.in]: I would drop this change.

File: doc/man1/openssl-s_client.pod.in [hunk 4]
```diff
@@ -843,7 +843,7 @@ If there are problems verifying a server certificate then the
 B<-showcerts> option can be used to show all the certificates sent by the
 server.
 
-This command is a test tool and is designed to continue the
+This command is for testing and is designed to continue the
```
Comment (t8m) [inline doc/man1/openssl-s_client.pod.in]: And this as well.

File: doc/man1/openssl-sess_id.pod [hunk 5]
```diff
@@ -22,7 +22,7 @@ B<openssl> B<sess_id>
 This command processes the encoded version of the SSL session
 structure and optionally prints out SSL session details (for example
 the SSL session master key) in human readable format. Since this is a
-diagnostic tool that needs some knowledge of the SSL protocol to use
+diagnostic aid that needs some knowledge of the SSL protocol to use
```
Comment (t8m) [inline doc/man1/openssl-sess_id.pod]: And here.

File: doc/man1/openssl-x509.pod.in [hunk 6]
```diff
@@ -80,7 +80,7 @@ B<openssl> B<x509>
 
 =head1 DESCRIPTION
 
-This command is a multi purpose certificate utility. It can
+This command can
```
Comment (t8m) [inline doc/man1/openssl-x509.pod.in]: I'd probably replace utility with tool and keep the sentence.

File: doc/man1/openssl.pod [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 =head1 NAME
 
-openssl - OpenSSL command line tool
+openssl - OpenSSL command line program
```
Comment (t8m) [inline doc/man1/openssl.pod]: Unsure here. Tool looks better to me but I do not insist.

**Comment by richsalz** [comment]
Re @t8m 's comment, https://github.com/openssl/openssl/pull/11123#pullrequestreview-369358017, I'm kinda fine with that.  In the past I've updated find-doc-nits so that errors cannot be repeated. In this PR it led to some collatoral damage, as it were.  I tried some wording contortions so that I could prevent "tool" or "utility" from appearing in the NAME section.  Let me see if I can fix that a better way.  I will undo the other-than-NAME section changes.

File: doc/man1/openssl-x509.pod.in (hunk 6 above)
Comment (richsalz) [reply doc/man1/openssl-x509.pod.in]: done.

File: doc/man1/openssl-asn1parse.pod (hunk 1 above)
Comment (richsalz) [reply doc/man1/openssl-asn1parse.pod]: yes, reverted.

File: doc/man1/openssl-s_client.pod.in (hunk 3 above)
Comment (richsalz) [reply doc/man1/openssl-s_client.pod.in]: reverted.

File: doc/man1/openssl.pod (hunk 7 above)
Comment (richsalz) [reply doc/man1/openssl.pod]: left this one.

**Comment by richsalz** [comment]
So I changed things to only complain about tool/utility in the NAME section.  Rebased, re-ordered some commits, and pushed.  Thanks for the feedback.

File: util/find-doc-nits [hunk 8]
```diff
@@ -601,6 +601,13 @@ sub wording {
     }
     err($id, "found 'epoch' should use 'Epoch'")
         if $contents =~ /\bepoch\b/;
+    if ( $id =~ m@man1/@ ) {
+        err($id, "found 'tool' in NAME, should use 'command'")
+            if $contents =~ /=head1 NAME.*tool.*=head1 SYNOPSIS/s;
+        err($id, "found 'utility' in NAME, should use 'command'")
+            if $contents =~ /NAME.*tool.*=head1 SYNOPSIS/m;
```
Comment (t8m) [inline util/find-doc-nits]: This is matching for tool and not utility also `/m` modifier does not look correct.

Comment (richsalz) [reply util/find-doc-nits]: yeah, incomplete edit.  fixup commit pushed, thanks!

**Comment by t8m** [comment]
Travis failure (doc nits) seems relevant.

**Comment by richsalz** [comment]
I don't get it the travis failures.  I don't see them no matter how hard i try.  (I had it before, and fixed it; I wonder if CI's got out of sync with this PR?)  I am going to close this, wait a bit, and re-open to restart the CI's.

**Comment by levitte** [comment]
Wow. That amount of "Found warnings" indicates there's something wrong with the warning checker (also, *it would be more constructive if the actual warnings were displayed *)

**Comment by richsalz** [comment]
>that amount of "found warnings"

Yes :(  Works on my system, but resetting the counts to zero, but apparently not on the CI machines.  I will do another approach using (sigh) temp files.

**Comment by richsalz** [comment]
Carl,

Thanks for your generous offer.  We seem to be all set.

**Comment by richsalz** [comment]
I pushed a local version of Pod::Checker.  I don't know how to do perl inheritance and override a method.  Oh well.

**Comment by richsalz** [comment]
rebased, squashed, pushed.

**Comment by levitte** [comment]
I feel *very* uncomfortable with having a modified Pod::Checker located in the middle of our util/perl stuff.

I understand the effort to get warnings checked, but it needs a rethink, and should probably be a separate PR.  That would make it much easier to approve *the rest* of this PR.

**Comment by richsalz** [comment]
Yeah, I felt dirty doing that.  I think the right thing to do is subclass it and replace the poderror method but that is more perl than I currently know.  Maybe I'll play around a bit.  For now, I reverted that commit and rebased.

**Comment by levitte** [comment]
> I think the right thing to do is subclass it and replace the poderror method but that is more perl than I currently know.

That's the approach I'd choose.

File: doc/man1/openssl-ca.pod.in [hunk 9]
```diff
@@ -156,7 +156,7 @@ Names and values of these options are algorithm-specific.
 
 The password used to encrypt the private key. Since on some
 systems the command line arguments are visible (e.g. Unix with
-the L<ps(1)> utility) this option should be used with caution.
+the L<ps(1)> command) this option should be used with caution.
```
Comment (levitte) [inline doc/man1/openssl-ca.pod.in]: I would rephrase that parenthesis, something like "(e.g. when using L<ps(1)> on Unix)"

File: doc/man1/openssl-x509.pod.in [hunk 10]
```diff
@@ -81,7 +81,7 @@ B<openssl> B<x509>
 
 =head1 DESCRIPTION
 
-This command is a multi purpose certificate utility. It can
+This command is a multi-purposes certificate utility. It can
```
Comment (levitte) [inline doc/man1/openssl-x509.pod.in]: s/utility/command/

File: util/find-doc-nits [hunk 11]
```diff
@@ -601,6 +601,13 @@ sub wording {
     }
     err($id, "found 'epoch' should use 'Epoch'")
         if $contents =~ /\bepoch\b/;
+    if ( $id =~ m@man1/@ ) {
+        err($id, "found 'tool' in NAME, should use 'command'")
+            if $contents =~ /=head1 NAME.*tool.*=head1 SYNOPSIS/s;
```
Comment (levitte) [inline util/find-doc-nits]: Might I suggest this regexp?  `/=head1 NAME.*\btool\b.*=head1 SYNOPSIS/`

`\b` matches the boundary between word and non-word

File: util/find-doc-nits [hunk 12]
```diff
@@ -601,6 +601,13 @@ sub wording {
...
+        err($id, "found 'utility' in NAME, should use 'command'")
+            if $contents =~ /NAME.*utility.*=head1 SYNOPSIS/s;
```
Comment (levitte) [inline util/find-doc-nits]: Same kind of change here

File: util/find-doc-nits (hunk 11 above)
Comment (richsalz) [reply util/find-doc-nits]: Worried someone might put, "stool" in the name? :)  Sure.

**Comment by richsalz** [comment]
Added a fixup commit with all of @levitte's suggestions.

File: doc/man1/openssl-ca.pod.in [hunk 13]
```diff
@@ -155,8 +155,9 @@ Names and values of these options are algorithm-specific.
 =for openssl foreign manual ps(1)
 
 The password used to encrypt the private key. Since on some
-systems the command line arguments are visible (e.g. Unix with
-the L<ps(1)> utility) this option should be used with caution.
+systems the command line arguments are visible (e.g., when using
+the L<ps(1)> command on Unix),
```
Comment (levitte) [inline doc/man1/openssl-ca.pod.in]: Lose "the" and "command"

File: doc/man1/openssl-ciphers.pod.in [hunk 14]
```diff
@@ -31,7 +31,7 @@ B<openssl> B<ciphers>
 =head1 DESCRIPTION
 
 This command converts textual OpenSSL cipher lists into
-ordered SSL cipher preference lists. It can be used as a test tool to
+ordered SSL cipher preference lists. It can be used when testing to
```
Comment (levitte) [inline doc/man1/openssl-ciphers.pod.in]: Lose "when testing"

File: doc/man1/openssl-ca.pod.in (hunk 13 above)
Comment (richsalz) [reply doc/man1/openssl-ca.pod.in]: updated fixup commit pushed to address both new issues.

**Comment by levitte** [comment]
Merged

4b5371913ed9bb4bfd8a55a30458932799296ab9 DOCS: Use "command" not "tool" or "utility"

**Comment by richsalz** [comment]
Thanks for the discussion and review.  Especially since this was nits about words, not where to place the curly braces :)
